proto/substrait/algebra.proto

```diff
@@ -32,6 +34,11 @@ message RelCommon {
     repeated int32 output_mapping = 1;
   }
 
+  message Metadata {
+    // Sets (or resets) the output field names for any relation.
+    repeated string output_names = 1;
```

Comment:
maybe just add this in Hint (since it's something a consumer can ignore), next to `string alias = 3` which serves a similar purpose?

Reply:
Either would work for me.  The only reason I avoided this was because of the comment:

>   // Changes to the operation that can influence efficiency/performance but
>   // should not impact correctness.

In the case of `ExpandRel` in Spark, it does impact the correctness.

Other than that, I'm happy to go with whatever the maintainers prefer :)

Reply:
Hm, how does it affect correctness? If the expected names are not supplied, you should be able to pass in some generated names, and the plan's correctness should be intact, right? (TBH I don't know how exactly Expand works so I might be missing something, but for all I know https://github.com/substrait-io/substrait/pull/661#issuecomment-2248511849 is correct in that only the names at the end of the plan should matter for correctness and all intermediary names can be whatever. (That said, for debugging/readability I do like having correct names also in the middle of the plan, so in general I'm totally for this change!)

Reply:
If the Spark implementation uses the names to connect up to other parts of the plan then it is using the new field for correctness.  If however it is only using them to ensure the roundtrip works (i.e. giving names to something that wouldn't otherwise be there) then it is metadata.  I would hope that the Spark implementation would use the second approach (and if it does it would fall under metadata).

Reply:
I've done some further testing, and I think I agree this is an artefact of the round-trip test mechanism that the `spark` module is using.
I've moved this into the `Hint` message, as suggested.